File: Codes in Python3/Dynamic Programming/Maximum sum of non-adjacent elements (Circular).py
class Solution:
    def helper(self, nums: List[int]) -> int:
        n=len(nums)
        prev2=0
        prev=nums[0]
        for index in range(n):
            pick = nums[index]
            if index>1:
                pick+=prev2
            notpick = 0 + prev
            curi = max(pick,notpick)
            prev2=prev
            prev=curi
        return prev    
    def rob(self, nums: List[int]) -> int:
        n=len(nums)
        if n==1:
            return nums[0]
        temp1=[]
        temp2=[]
        for i in range(n):
            if i!=0:
                temp1.append(nums[i])
            if i!=(n-1):
                temp2.append(nums[i])
        return max(self.helper(temp1),self.helper(temp2))  

File: Codes in C++/Dynamic Programming/Maximum sum of non-adjacent elements (Circular).cpp
class Solution {
public:
    long long int helper(vector<int>& nums) {
    int prev2=0;
    int prev=nums[0];
    int curi=-1;
    int n=nums.size();
    for(int i=1;i<n;i++)
    {
        int pick = nums[i];
        if(i>1)
            pick+=prev2;
        int not_pick = 0 + prev;
        curi=max(pick,not_pick);
        prev2=prev;
        prev=curi;
    }
    return prev;
    }
    long long int rob(vector<int>& nums) {
     int n=nums.size();
     vector<int> temp1,temp2;
     if(n==1) return nums[0];
     for(int i=0;i<n;i++)
     {
         if(i!=0)
            temp2.push_back(nums[i]);
         if(i!=(n-1))
            temp1.push_back(nums[i]);
     }
     return max(helper(temp1),helper(temp2));
    }
};